flow/flow.go

```diff
@@ -893,17 +893,27 @@ func (f *Flow) newApplicationLayer(packet *Packet, opts Opts) error {
 	if (opts.ExtraLayers & DNSLayer) != 0 {
 		if layer := packet.Layer(layers.LayerTypeDNS); layer != nil {
 			d := layer.(*layers.DNS)
+			answers := make([]string, len(d.Answers))
+			questions := make([]string, len(d.Questions))
+			for _, v := range d.Answers {
```

Comment:
it should be more something like this : 

```
                      if len(d.Answers) > 0 {
                               answers := make([]string, len(d.Answers))
                               for i, v := range d.Answers {
                                       answers[i] = string(v.IP)
                               }
                               f.DNS.Answer = answers
                        }
```

placed after `f.DNS = &fl.DNS{....`

Reply:
Since go 1.10.1 there should not be any significant difference in performance and using append is more idiomatic 

https://stackoverflow.com/questions/38654729/golang-slice-append-vs-assign-performance

Reply:
Using append it appends an empty entry due to the make/len. That said my point was about adding only if needed (if statement).

Reply:
done